replication.drush.inc

```diff
@@ -0,0 +1,261 @@
+      $options = array_merge($options, $url_parts);
+    }
+    $client = CouchDBClient::create($options);
+    $continuous = drush_get_option('continuous');
+    return $client->replicate($source, $target, NULL, $continuous);
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_replication_stop_replication($source, $target) {
+  try {
+    $options = array('dbname' => '');
+    if ($replicator = drush_get_option('replicator')) {
+      $url_parts = _get_url_parts($replicator, TRUE);
+      $options = array_merge($options, $url_parts);
+    }
+    $client = CouchDBClient::create($options);
+    return $client->replicate($source, $target, TRUE, TRUE);
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ *
+ * Prints information about the specific active replication between target and
+ * source databases.
+ */
+function drush_replication_active_replication($source = NULL, $target = NULL) {
+  try {
+    $options = array('dbname' => '');
+    if ($replicator = drush_get_option('replicator')) {
+      $url_parts = _get_url_parts($replicator, TRUE);
+      $options = array_merge($options, $url_parts);
+    }
+    $client = CouchDBClient::create($options);
+    $results = $client->getActiveTasks();
+    foreach ($results as $key => $result) {
+      $results[$key]['started_on'] = date('D, j M Y, H:i:s e', $result['started_on']);
+      if ($source && $target && is_array($results)) {
+        $source_diff = array_diff(_get_url_parts($result['source']), _get_url_parts($source));
+        $target_diff = array_diff(_get_url_parts($result['target']), _get_url_parts($target));
+        if (empty($source_diff) && empty($target_diff)) {
+          // Return information about one active replication.
+          return array($results[$key]);
+        }
+        else {
+          drupal_set_message('No active replication.', 'status');
+          return;
+        }
+      }
+    }
+    if (!empty($results)) {
+      // Return information about all active replications.
+      return $results;
+    }
+    else {
+      drupal_set_message('No active replications.', 'status');
+    }
+  }
+  catch (\Exception $e) {
+    drush_set_error($e->getMessage());
+  }
+}
+
+/**
+ * Helper function for command validation.
+ *
+ * @param $params
+ */
+function _replication_command_validate($params) {
+  if (_get_http_response_code('http://localhost:5984') != 200) {
+    drush_set_error(dt('CouchDB not found. Please install CouchDB.'));
+  }
+  if (_get_http_response_code($params['0']) != 200) {
+    drush_set_error(dt('Source database not found.'));
+  }
+  if (_get_http_response_code($params['1']) != 200) {
+    drush_set_error(dt('Target database not found.'));
+  }
+}
+
+/**
+ * Returns the response code for a request.
+ *
+ * @param $url
+ *
+ * @return string
+ */
+function _get_http_response_code($url) {
```

Comment:
We should namespace this function properly, i.e. the function name needs to start with either `replication_*` or `_replication_*`.